### PR TITLE
build: improve Makefile setup on macOS

### PR DESCRIPTION
**Pull request by behnazh-w**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -134,9 +141,15 @@ souffle:
                 sudo apt update; \
                 sudo apt install souffle; \
                 ;; \
+			"Darwin") \
```
Comment (tromai) [inline Makefile:144]: I checked on my machine and it seems that this block for `"Darwin"` is indented using tabs instead of spaces.

Comment (behnazh-w) [reply Makefile]: See commit [1cdef4e](https://github.com/oracle/macaron/pull/383/commits/1cdef4e8fde94b7fa027149483caf90ea9441a80)

File: src/macaron/config/defaults.py [hunk 2]
```diff
@@ -163,9 +163,6 @@ def create_defaults(output_path: str, cwd_path: str) -> bool:
             os.path.relpath(os.path.join(output_path, "defaults.ini"), cwd_path),
         )
         return True
-    except shutil.Error as error:
+    except (OSError, shutil.Error) as error:
```
Comment (tromai) [inline src/macaron/config/defaults.py:166]: Should we put a small comment above this line saying why we added the OSError here?

Comment (behnazh-w) [reply src/macaron/config/defaults.py:167]: See commit [dc4a44d](https://github.com/oracle/macaron/pull/383/commits/dc4a44dc54e47d3e09ac886fad768f327802da57)

**Review by tromai — APPROVED** [review]

**Pull request merged**